packages/core/strapi/lib/utils/machine-id.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const { machineIdSync } = require('node-machine-id');
+const uuid = require('uuid');
+
+module.exports = () => {
+  try {
+    const deviceId = machineIdSync();
+    return deviceId;
+  } catch (error) {
+    const deviceId = uuid();
```

Comment:
I think `machineIdSync()` return the same value each time Strapi is started. But `uuid()` will return a different one. So it would create confusion in our telemetry. Is there another way to get unique identified of the machine? Or maybe storing the uuid somewhere but (1) some systems are read only so it would break for them (2) not sure storing it in DB is ideal

Reply:
Sorry Pierre, I didn't realise you're out of the loop on the conversation about this issue. My bad. 

You're correct, machineIdSync() returns the same value, as the value is written into the OS at install time, but a shell is needed to retrieve it. So, this package won't work on any hosts that does not have a shell and I did not find another good alternative. I was speaking to Alex about this and apparently we anyway exclude Docker analytics, so if we have UUID instead of machineIdSync(), the damage will be minimal. The original suggestion was to return `null` or `undefined` if machineIdSync() throws an error, but I figured that would a) skew analytics results even more (with uuid as deviceId we can still trace all analytics calls within one container session, with undefined we would see all distroless container analytics as one lump); and b) we need some sort of deviceId for our tracker to work correclty (with null or undefined values it's not going to work as intended).

Also note that this fix only concerns non-shell hosts, so in an average Docker container machineIdSync() will still work as expected, as the image will most likely be based on some OS.

Reply:
Thank you for the explanation! So on the analytic side, it is not too much trouble to have a different `deviceId` each time strapi is restarted?

Reply:
No worries, you are welcome :) Correct, it's not too much trouble for the moment, we don't have an alternative anyway and it's a limited use case as well. I'll mark this conversation as resolved for now, but I will also add @Aurelsicoko as a reviewer just to confirm